### PR TITLE
add shape propagate pass on mhlo

### PR DESCRIPTION
**Pull request by Yancey1989**
```python
import torch
import torch.nn as nn
import unittest
import torch_xla.core.xla_model as xm
import torch.nn.functional as F

torch.manual_seed(1001)
xla_device = xm.xla_device()
#xla_device = "cuda"
dtype = torch.float16

class TestMLP(nn.Module):
  def __init__(self):
    super(TestMLP, self).__init__()
    self.fc = nn.Linear(1024, 4096, bias=False)

  def forward(self, x):
    return self.fc(x)

class TestDynamicShape(unittest.TestCase):
  def test_embeded(self):
    device = xla_device

    m = TestMLP().to(device)
    m.train()
    with torch.cuda.amp.autocast(dtype=torch.bfloat16):
      for i in range(10):
        xla_tensor = torch.rand(i+1, 1024, requires_grad=False)
        xla_tensor = xla_tensor.to(device)

        xla_output = m(xla_tensor)
        loss = xla_output.sum()
        loss.backward()
        xm.mark_step()
        print(f"iteration :{i}, loss: {loss.clone().detach()}")
if __name__ == '__main__':
  unittest.main()
```

I got the same loss value with the CUDA device using the above simple MLP module.

### CODE REVIEW

**Review by eedalong — APPROVED** [review]
LGTM
